### PR TITLE
Enable clustering for gene set heatmaps 

### PR DESCRIPTION
**Pull request by fedde-s**
List any queried gene set profile in the heatmap menu, displaying only
the clustering button as it's queried via the query page and the ‘Add
Genes’ and ‘Remove Heatmap’ buttons do not apply to it.

Collapse expansion tracks before clustering a heatmap, as these are
generally not of the same data type as the main tracks of the heatmap,
and including them when clustering would not make sense.

## Screenshot ##

Screen shot showing an Oncoprint with a clustered gene set heatmap and a heatmap menu containing only the cluster button for this profile

### CODE REVIEW

File: src/shared/components/oncoprint/DeltaUtils.ts [hunk 1]
```diff
@@ -242,7 +242,11 @@ export function transitionSortConfig(
         nextSortConfig.type === "order" &&
         (prevSortConfig.order !== nextSortConfig.order))
         ||
-        !_.isEqual(prevSortConfig, nextSortConfig)) {
+        !_.isEqual(prevSortConfig, nextSortConfig)
+    ) {
+        if (nextSortConfig.type === "cluster") {
+            oncoprint.removeAllExpansionTracksInGroup(nextSortConfig.track_group_index);
```
Comment (fedde-s) [inline src/shared/components/oncoprint/DeltaUtils.ts]: @adamabeshouse, TypeScript claims that this method does not exist, even though it's defined here: https://github.com/cBioPortal/oncoprintjs/blob/v1.0.79/src/js/oncoprint.js#L551

Could you point me at where I should declare the type for it?

Comment (adamabeshouse) [reply src/shared/components/oncoprint/DeltaUtils.ts]: ah, yes you'll need to add it into index.d.ts in the home directory of oncoprintjs

Comment (fedde-s) [reply src/shared/components/oncoprint/DeltaUtils.ts]: Done in cBioPortal/oncoprintjs#59

Comment (adamabeshouse) [reply src/shared/components/oncoprint/DeltaUtils.ts]: released `v1.0.83`

**Comment by fedde-s** [comment]
Superseded by cBioPortal/cbioportal-frontend#982

**Pull request closed without merging**